packages/fx-core/src/component/driver/middleware/addSpecialScriptTelemetry.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { HookContext, Middleware, NextFunction } from "@feathersjs/hooks";
+import { WrapDriverContext } from "../util/wrapUtil";
+import {
+  TeamsFxTelemetryConfig,
+  TeamsFxTelemetryReporter,
+} from "../../utils/teamsFxTelemetryReporter";
+import { TelemetryConstant } from "../../constant/commonConstant";
+import { performance } from "perf_hooks";
+import { TelemetryConstants } from "../../constants";
+import { isExecutionResult } from "./addStartAndEndTelemetry";
+import { maskSecretValues } from "../../utils/envUtil";
+
+/**
+ * A special telemetry middleware for script driver running for a SWA deploy.
+ * @param eventName The event name to be sent.
+ */
+export function addSpecialScriptTelemetry(eventName: string): Middleware {
```

Comment:
I think this function is duplicate with the `addStartAndEndTelemetry` function in src/component/driver/middleware.
Is it possible to reduce the duplicate by some tricks?
I don't think the following code works, but I think there may be an approach to do similar thing:
```
export function addSpecialScriptTelemetry(eventName: string): Middleware {
    const telemetryMW= addStartAndEndTelemetry(eventName, componentName);
    return {
        doSomethingSpecial();
        telemetryMW();
    }
}
```

Reply:
I think there is no redundant code here. I just constructed another telemetry data and send it.

Maybe all the same code is `await next();` ?